src/model.ts

```diff
@@ -694,7 +694,14 @@ export class SpecQueryModel {
 
         // otherwise, assign the proper to the field def
         if (encQ[prop] !== undefined) {
-          fieldDef[prop] = encQ[prop];
+
+          if (prop === Property.AXIS || prop === Property.LEGEND || prop === Property.SCALE || prop === Property.SORT) {
```

Comment:
This condition is surely poorly written as both cases and if and else contain the same statements.   This is hard to read.  Why don't you combine the predicate in a better way?